api/environments/models.py

```diff
@@ -88,6 +90,21 @@ def create_feature_states(self):
                 else feature.default_enabled,
             )
 
+    @property
+    def last_updated_at(self) -> typing.Optional[datetime]:
+        # NOTE: `last_updated_at` is a proxy for changes that affect a value of a given flag in the environment
+        last_audit_log = (
+            AuditLog.objects.exclude(
+                related_object_type=RelatedObjectType.CHANGE_REQUEST.name
+            )
+            .filter(Q(environment=self) | Q(project=self.project))
+            .order_by("created_date")
+            .first()
+        )
+        # last_audit_log should never be None unless the environment was created manually
+        # and no audit logs were created for it
+        return last_audit_log.created_date if last_audit_log else None
```

Comment:
@matthewelwell I am not sure if we should raise an exception here if no audit logs exists for the environment, what do you think?

Reply:
Couldn't we add a `created_at` field to the environment and use that in that case? 

I think it would be as simple as `Environment.objects.filter(id=self.environment_id).update(updated_at=timezone.now())` right?